### PR TITLE
Upgraded Code Base to 2x

### PR DESCRIPTION
**Pull request by InsaneNaman**
Upgraded the all the src file to 2x version.

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
Awesome!  Will give this a try!  Thanks for opening the PR!  :smile:

**Comment by InsaneNaman** [comment]
Your welcome sir . Waiting for your feedback. :smile: and yeah i'll create another PR as soon as i remove all the warnings and if i find any bug .

File: src/rk-events-core.html [hunk 1]
```diff
@@ -8,156 +8,167 @@
...
-            if (Immutable.is(this._immutableMap, msgs)) {
-              console.log("IMMUTABLE equal;");
-              return;
-            }
-          } else {
-            this.set('_immutableMap', msgs);
-            console.log("IMMUTABLE null");
-            return;
+    addMessages(msgs, instance) {
+      console.log("CCCCCAALEEDDD");
```
Comment (karlprieb) [inline src/rk-events-core.html]: Can you remove this `console.log`? :)

**Review by karlprieb — CHANGES_REQUESTED** [review]

**Comment by karlprieb** [comment]
@InsaneNaman I cant make the bundled build. There is some special step on the build that I need to do?

File: src/rk-events-core.html (hunk 1 above)
Comment (InsaneNaman) [reply src/rk-events-core.html]: yes sir for sure, it was pre written that's why i didn't remove it .But now in the latest commit i have done that . Please have a look. I hope i have done it in the right manner (i.e i did changes on my machine then pushed to the respected remote branch ). If i have done something in the wrong way do let me know . :smile:

**Comment by InsaneNaman** [comment]
@karlprieb  as far as i know . No there is not such special step but i think so 
1-You will need to update your polymer-cli ( as mentioned in their latest [documentation](https://www.polymer-project.org/2.0/docs/upgrade#before-you-begin) using this command :

npm update polymer-cli@next

2-bower install 

3-polymer build 

4-polymer serve

Please do let me know if it's working or not

**Comment by Sing-Li** [comment]
@InsaneNaman   - any chance of fixing the build warnings?   Can you explain what causes them?  TIA

```
warn:    Unable to optimize .js file /Users/sam/dev/github/Rocket.Chat.PWA/src/rk-room-view.html_script_0.js 
{ err: 
   { message: 'Unexpected token: name (tp)',
     filename: 0,
     line: 12,
     col: 12,
     pos: 231 } }
warn:    Unable to optimize .js file /Users/sam/dev/github/Rocket.Chat.PWA/src/rk-events-core.html_script_0.js 
{ err: 
   { message: 'Unexpected token: operator (>)',
     filename: 0,
     line: 58,
     col: 48,
     pos: 1380 } }
info:    Build complete!
```

In addition, can you hook it up to a  *live*  Rocket.Chat server instance (and actually fetch and display the fetched messages) ?

**Comment by InsaneNaman** [comment]
@Sing-Li Sir , though in my PR there are some warnings(and most of the warnings are indentation type and unused variable/methods ,that i didn't remove because i thought we can use it in future ) but i am not getting this type of build error anywhere :( . I just once again did a build of your official repo and the final PR that i sent  but in both cases i am not getting any of these error. I am attaching the screenshot . 
and just by looking at your ss i only can say there is extra unnecessary token is present in the 0.js file that polymer made itself.  

screenshot from 2017-03-18 21-37-46
screenshot from 2017-03-18 21-27-59

So please can you tell me more about this build error. And yeah i can hook it up to a live server just please provide me some documentation and other useful links . I would be happy to do that.

**Comment by InsaneNaman** [comment]
I would suggest you to please once again update your polymer cli using (if you haven't done ) 
npm update polymer-cli@next
and if required bower and npm too. :smile: 

because this error may be occurring due to any outdated dependency .

**Comment by Sing-Li** [comment]
@InsaneNaman  -- see this line ... this is how to hookup a *live* instance of Rocket.Chat server ...   https://github.com/RocketChat/Rocket.Chat.PWA/blob/develop/src/rk-room-view.html#L78

You can get that going while we figure out how to build your PR.

**Comment by InsaneNaman** [comment]
Sure sir , do let me know if you need any help from my side in building my PR. And i look over the web hook stuff . Give me some time to go through the link.

**Pull request closed without merging**